google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -261,6 +272,21 @@ protected String defaultHost() {
   }
 
   protected String defaultProject() {
+    return defaultProjectId();
+  }
+
+  /**
+   * Returns the default project ID, or {@code null} if no default project ID could be found. This
+   * method returns the first available project ID among the following sources:
+   * <ol>
+   *   <li>App Engine project ID
```

Comment:
It seems as though the order here is wrong, we seem to get it from PROJECT_ENV_NAME and then LEGACY_PROJECT_ENV_NAME both for gcloud, then its AppEngine, then JSON credentials from service account, then gcloud configuration.

Where does the compute engine project id come from?